admin/includes/classes/AdminRequestSanitizer.php

```diff
@@ -555,6 +555,45 @@ private function filterMultiDimensional($parameterName, $parameterDefinition)
 
     /**
      * @param $parameterName
+     * @param $parameterDefinition
+     */
+    private function filterSimpleArray($parameterName, $parameterDefinition)
+    {
+        $requestPost = $_POST;
+        if (!isset($requestPost[$parameterName])) {
+            return;
+        }
+        $this->debugMessages[] = 'PROCESSING SIMPLE_ARRAY == ' . $parameterName;
+
+        $hacked = $requestPost[$parameterName];
+        foreach ($hacked as $pkey => $pvalue) {
+            if (isset($parameterDefinition['params'][$pkey])) {
+                unset($requestPost[$parameterName][$pkey]);
+                unset($_POST);
+                $_POST[$pkey] = $pvalue;
+                $type = $parameterDefinition['params'][$pkey]['sanitizerType'];
+                $params = isset($parameterDefinition['params'][$pkey]['params']) ? $parameterDefinition['params'][$pkey]['params'] : null;
+                $newParameterDefinition = array('sanitizerType' => $type, 'params' => $params);
+                $this->runSpecificSanitizer($pkey, $newParameterDefinition);
+                $requestPost[$parameterName][$pkey] = $_POST[$pkey];
+                if ($type != 'STRICT_SANITIZE_VALUES' && $type != 'STRICT_SANITIZE_KEYS') {
+                    $this->postKeysAlreadySanitized[] = $pkey;
+                }
+            }
+            if ($this->doStrictSanitization) {
```

Comment:
The code in this area looks like it could use some additional work to fully incorporate STRICT_SANITIZE_KEYS. 

Ie. `unset` the associated key in the `$requestPost` variable up front, then encapsulate the assignment in an `if` statement to see if `$key` is set or exists in `$_POST`.